grails-app/services/au/org/ala/ecodata/MetadataService.groovy

```diff
@@ -135,8 +135,32 @@ class MetadataService {
         // Remove deprecated activities
         activities = activities.findAll {!it.status || it.status == ACTIVE}
 
-        Map byCategory = [:]
+        groupActivities(activities)
+    }
+
+    /**
+     * Returns a Map of activity types grouped by category.
+     * @param programId Most of the legacy projects are updated recently to insert programId (see github #2741)
+     * @return a Map, key: String, value: List of name, description for each activity in the category
+     */
+    Map activitiesListByProgramId(String programId) {
+        Map config = Program.findByProgramId(programId)?.config
+        List<String> activities = config.activities.collect{it.name}
 
+        List<ActivityForm> forms = activityFormService.search(category:activities)
+        List results = forms.collect {[name:it.name, category:it.category, type:it.type, description:it.description, status:it.status]}
+        results = results.findAll {!it.status || it.status == ACTIVE}
```

Comment:
Why is this status based filtering necessary?  If there is bad data in the database (for the status field) we should fix that rather than do this workaround.